src/ray/core_worker/transport/direct_actor_task_submitter.cc

```diff
@@ -286,19 +286,8 @@ void CoreWorkerDirectActorTaskSubmitter::DisconnectActor(
             task_id, error_type, &status, &error_info));
       }
 
-      auto &wait_for_death_info_tasks = queue->second.wait_for_death_info_tasks;
-
-      RAY_LOG(INFO) << "Failing tasks waiting for death info, size="
-                    << wait_for_death_info_tasks.size() << ", actor_id=" << actor_id;
-      for (auto &net_err_task : wait_for_death_info_tasks) {
-        RAY_UNUSED(task_finisher_.MarkTaskReturnObjectsFailed(
-            net_err_task.second, error_type, &error_info));
-      }
-
-      // No need to clean up tasks that have been sent and are waiting for
-      // replies. They will be treated as failed once the connection dies.
-      // We retain the sequencing information so that we can properly fail
-      // any tasks submitted after the actor death.
+      // We need to execute this outside of the lock to prevent deadlock.
+      wait_for_death_info_tasks = std::move(queue->second.wait_for_death_info_tasks);
```

Comment:
should we clear `queue->second.wait_for_death_info_tasks`? I'm under the impression the object being moved aways's state is undefined.